### PR TITLE
Created Snapshotted interface

### PR DESCRIPTION
**Pull request by aoaaceai**
And also made the player, deck and game snapshotted

### CODE REVIEW

**Comment by secminhr** [comment]
Sorry, I couldn't see the difference between PR #2 and this PR?

**Comment by aoaaceai** [comment]
it should be solved now
sorry for messing up the version control.

Also changed the Card interface into CardUtil, then added a Card abstract class.

One TODO after this: redesign the snapshot stuff.

File: src/eod/card/abstraction/CardUtil.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package eod.card.abstraction;
+
+import eod.Deck;
+
+public interface CardUtil {
```
Comment (secminhr) [inline src/eod/card/abstraction/CardUtil.java]: Suggest rename to ```ICard```, representing Card Interface.

**Review by secminhr — COMMENTED** [review]
Please fix the conflicting files(show below), and it'll be ready to be merged.

**Review by secminhr — DISMISSED** [review]

**Comment by aoaaceai** [comment]
This one should be replaced with #5

**Pull request closed without merging**